core/worker.go

```diff
@@ -271,6 +275,21 @@ func newWorker(config *Config, chainConfig *params.ChainConfig, db ethdb.Databas
 	return worker
 }
 
+func (w *worker) setCoinbases(primaryAddress common.Address, secondaryAddress common.Address) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	// Use the MinerPreference to bias the decision
+	biasedFloat := math.Pow(rand.Float64(), (1-w.config.MinerPreference)*2)
```

Comment:
this wont get us 50-50, parabolic functions are not linear, so you cant approximate into getting 50-50 preference

Reply:
I looked it up, rand.Float64() already gives a value between 0 and 1, so we dont need anything else

Reply:
Float64 gives a uniform distribution between 0 and 1, but we want to integrate miner preference to bias the distribution. What do you suggest?

Reply:
Simplified it, just checking the threshold now